pkg/synthetictests/duplicated_events_special_test.go

```diff
@@ -254,3 +254,186 @@ func Test_testErrorUpdatingEndpointSlices(t *testing.T) {
+					To:   time.Unix(1, 0),
+				},
+			}
+			junit_tests := testFailedScheduling(e)
+			switch tt.kind {
+			case "pass":
+				if len(junit_tests) != 1 {
+					t.Errorf("This should've been a single passing test, but got %d tests", len(junit_tests))
+				}
+				if len(junit_tests[0].SystemOut) != 0 {
+					t.Errorf("This should've been a pass, but got %s", junit_tests[0].SystemErr)
+				}
+			case "fail":
+				if len(junit_tests) != 1 {
+					t.Errorf("This should've been a single failing test, but got %d tests", len(junit_tests))
+				}
+				if len(junit_tests[0].SystemOut) == 0 {
+					t.Error("This should've been a failure but got no output")
+				}
+			case "flake":
+				if len(junit_tests) != 2 {
+					t.Errorf("This should've been a two tests as flake, but got %d tests", len(junit_tests))
+				}
+			default:
+				t.Errorf("Unknown test kind")
+			}
+
+		})
+	}
+}
+
+func Test_testOperatorStatusChanged(t *testing.T) {
+	tests := []struct {
+		name    string
+		message string
+		kind    string
+	}{
+		{
+			name:    "pass",
+			message: "reason/OperatorStatusChanged Status for clusteroperator/etcd changed: Degraded message changed from \"NodeControllerDegraded: All master nodes are ready\nEtcdMembersDegraded: 2 of 3 members are available, ip-10-0-217-93.us-west-1.compute.internal is unhealthy\" to \"NodeControllerDegraded: All master nodes are ready\nEtcdMembersDegraded: No unhealthy members found\" (2 times)",
+			kind:    "pass",
+		},
+		{
+			name:    "flake",
+			message: "reason/OperatorStatusChanged Status for clusteroperator/etcd changed: Degraded message changed from \"NodeControllerDegraded: All master nodes are ready\nEtcdMembersDegraded: 2 of 3 members are available, ip-10-0-217-93.us-west-1.compute.internal is unhealthy\" to \"NodeControllerDegraded: All master nodes are ready\nEtcdMembersDegraded: No unhealthy members found\" (24 times)",
+			kind:    "flake",
+		},
+		{
+			name:    "flake 21",
+			message: "reason/OperatorStatusChanged Status for clusteroperator/etcd changed: Degraded message changed from \"NodeControllerDegraded: All master nodes are ready\nEtcdMembersDegraded: 2 of 3 members are available, ip-10-0-217-93.us-west-1.compute.internal is unhealthy\" to \"NodeControllerDegraded: All master nodes are ready\nEtcdMembersDegraded: No unhealthy members found\" (21 times)",
+			kind:    "flake",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := monitorapi.Intervals{
+				{
+					Condition: monitorapi.Condition{
+						Message: tt.message,
+					},
+					From: time.Unix(1, 0),
+					To:   time.Unix(1, 0),
+				},
+			}
+			junit_tests := testOperatorStatusChanged(e)
+			switch tt.kind {
+			case "pass":
+				if len(junit_tests) != 1 {
+					t.Errorf("This should've been a single passing test, but got %d tests", len(junit_tests))
+				}
+				if len(junit_tests[0].SystemOut) != 0 {
+					t.Errorf("This should've been a pass, but got %s", junit_tests[0].SystemErr)
+				}
```

Comment:
The if is checking SystemOut, but then looking at SystemErr - is that intentional?

Reply:
I tried to keep it similar to other unit tests above